src/devices/Pca9685/Pca9685.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Device.I2c;
+using System.Threading;
+using static Iot.Device.Pca9685.Register;
+using static Iot.Device.Pca9685.Bit;
+
+namespace Iot.Device.Pca9685
+{
+    /// <summary>
+    /// PCA9685 PWM LED/servo controller
+    /// </summary>
+    public class Pca9685 : IDisposable
+    {
+
+        /// <summary>
+        /// I2C Device
+        /// </summary>
+        public I2cDevice Device { get; private set; }
+
+        /// <summary>
+        /// Get default clock rate. Set if you are using external clock.
+        /// </summary>
+        public double ClockRate { get; set; } = 25000000;   // 25MHz
+
+        /// <summary>
+        /// Set PWM frequency or get effective value.
+        /// </summary>
+        public double PwmFrequency
+        {
+            get => _pwmFrequency;
+            set
+            {
+                Prescale = GetPrescale(value);
+            }
+        }
+
+        /// <summary>
+        /// Set PWM frequency using prescale value or get the value.
+        /// </summary>
+        public byte Prescale
+        {
+            get => _prescale;
+            set
+            {
+                var v = value < 3 ? (byte)3 : value;  // min value is 3
+                SetPwmFrequency(v);
+                _prescale = v;
+                _pwmFrequency = GetFreq(v);
+            }
+        }
+
+        private double _pwmFrequency;
+        private byte _prescale;
+
+        /// <summary>
+        /// Initialize PCA9685
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device to be used</param>
+        public Pca9685(I2cDevice i2cDevice)
+        {
+            // Setup I2C interface for the device.
+            Device = i2cDevice;
+
+            SetPwm(0, 0);
+
+            Span<byte> buffer = stackalloc byte[2];
+
+            buffer[0] = (byte)MODE2;
```

Comment:
try this 😀
```c#
Span<byte> buffer = stackalloc byte[2] { (byte)MODE2, (byte)OUTDRV };

Reply:
I have a question - enum Register extends byte. Why I need explicitly convert MODE2 to byte when putting it into a byte array?

Reply:
if you don't extend byte, the code will look like this
```C#
public enum Register : int
{

}
```

Reply:
but why enum:byte to byte needs **explicit** conversion?

Reply:
@HumJ0218 this is a good question and I find this really annoying as well, might be worth asking this question here: https://github.com/dotnet/roslyn/issues

Reply:
@krwq I found it in .NET document that I really need an explicit conversion.

https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/enum#example-1

    enum Range : long { Max = 2147483648L, Min = 255L };
    static void Main()
    {
        long x = (long)Range.Max;
        ...
    }